supernode.cpp

```diff
@@ -67,8 +67,18 @@ void Supernode::batch_update(const std::vector<vec_t>& updates) {
    * Considered using spin-threads and parallelism within sketch::update, but
    * this was slow (at least on small graph inputs).
    */
-  #pragma omp parallel for num_threads(GraphWorker::get_group_size())
+  std::vector<Sketch*> deltas(logn);
+  #pragma omp parallel for num_threads(GraphWorker::get_group_size()) default(none) shared(deltas, updates)
   for (unsigned i = 0; i < sketches.size(); ++i) {
-    sketches[i]->batch_update(updates);
+    deltas[i] = new Sketch(*sketches[i]);
```

Comment:
Won't this cause problems? Say we start with a sketch which has updates for nodes = a,b,c in it. Then we apply a update which deletes node b. So our delta is now = a,c. When we below add this delta to the node sketch we will end up with a final answer of nodes = b which is wrong. The answer should be nodes = a,c

Reply:
Additionally this is worrisome because we might make these copies in the middle of another node applying the deltas. So we would end up with some wonky results.

for this reason and the above concern we should start with null sketches rather than copies.

Reply:
Marking necessary for merge

Reply:
The copy-constructor creates a blank sketch with the same meta-fields as the original. That is, `bucket_a` and `bucket_c` are zeroed-out, but the rest is the same.

Reply:
Although now that you mention it, is there a case where we want an actual copy-constructor?

Reply:
Gotcha

Reply:
Maybe at some point down the road but for now copying sketches is bad.